src/include/btmem.h

```diff
@@ -960,15 +946,71 @@ struct __wt_ref {
-     * field is freed and replaced by the WT_REF.update array (needed for subsequent transaction
-     * commit/abort). Doing anything other than testing if WT_REF.del/update is non-NULL (which
-     * eviction does), requires the WT_REF be locked. If the locked WT_REF's previous state was
-     * WT_REF_DELETED, WT_REF.del is valid, if the WT_REF's previous state was an in-memory state,
-     * then WT_REF.update is valid.
+     * Fast-truncate information, written-to/read-from disk as necessary in the internal page's
+     * deleted page proxy cell. When a WT_REF first becomes part of a fast-truncate operation, the
+     * ft_info.del field is allocated and initialized.
+     *
+     * Fast-truncate pages might have to be instantiated if a thread for which the operation isn't
+     * visible accesses the page. This can happen if the operation hasn't committed yet; it can also
+     * happen if an older read transaction visits the page; and it can happen if the fast-truncate
+     * operation is included in a checkpoint and then seen later, after a restart or via a
+     * checkpoint cursor.
+     *
+     * If the page must be instantiated for any reason: (1) WT_UPDATE structures are created for the
+     * page entries, (2) the transaction information from ft_info.del is copied to those WT_UPDATE
+     * structures (making them a match for the truncate operation), (3) the ft_info.del field is
+     * discarded, and (4) the WT_REF state switches to WT_REF_MEM.
+     *
+     * If the fast-truncate operation has not yet committed, additionally the ft_info.update field
+     * is created, which is an array of references to the WT_UPDATE structures, for subsequent
+     * transaction commit/abort. (The page can split, so there needs to be some way to find all of
+     * the update structures.)
+     *
+     * Doing anything other than testing if ft_info.del or ft_info.update is non-NULL (which
+     * eviction does) requires the WT_REF be locked.
+     *
+     * Because ft_info is a union it is important to always access the correct field. It is also
+     * vital to interpret the state correctly and consider all the possible cases.
+     *
+     * The union access should be ft_info.del if the state is WT_REF_DELETED (states 1 and 2 below),
+     * and should be ft_info.update if the state is WT_REF_MEM (states 5-6 below). Otherwise,
+     * neither field is valid and the pointer should always be NULL.
+     *
+     * These are the possible states:
+     *
+     * 1. The WT_REF state is WT_REF_DELETED and ft_info.del is NULL. This means the page is deleted
+     * and the deletion is globally visible. Any on-disk page has been or will be discarded.
+     *
+     * 2. The WT_REF state is WT_REF_DELETED and ft_info.del is not NULL. The page is on disk, and
+     * is deleted, but the deletion may not yet be globally visible (or visible to any given reader
```

Comment:
The phrase "page is on disk, and is deleted" is confusing, because it sounds like the on-disk page is deleted. Maybe just get rid of "is on disk", I don't think it's needed.

Reply:
I think it should be clear that the page is not in memory; that was the idea. Let me try again... (updated in [42d20ae](https://github.com/wiredtiger/wiredtiger/pull/7909/commits/42d20ae234f880c06e2cc2133a5c772b2c7414a6))